apps/sms/test/marionette/conversion_banner_test.js

```diff
@@ -0,0 +1,114 @@
+  var apps = {};
+
+  var client = marionette.client({
+    prefs: {
+      'focusmanager.testmode': true
+    },
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var messagesApp;
+
+  function assertIsDisplayed(element) {
+    assert.isTrue(element.displayed(), 'Element should be displayed');
+  }
+
+  function assertIsNotDisplayed(element) {
+    assert.isFalse(element.displayed(), 'Element should not be displayed');
+  }
+
+  function waitForBannerToDisappear() {
+    client.helper
+      .waitForElementToDisappear(messagesApp.Composer.conversionBanner);
+  }
+
+  setup(function() {
+    messagesApp = Messages.create(client);
+
+    client.contentScript.inject(
+      __dirname + '/mocks/mock_navigator_moz_icc_manager.js'
+    );
+    client.contentScript.inject(
+      __dirname + '/mocks/mock_navigator_moz_mobile_message.js'
+    );
+  });
+
+  suite('Message Type Conversion Banner Test Suite', function() {
+
+    setup(function() {
+      messagesApp.launch();
+    });
+
+    test('Old conversion banner is cleared before entering a thread',
+    function() {
+      var threadList = messagesApp.ThreadList;
+
+      // Create a new message
+      threadList.navigateToComposer();
+      // Force it to be MMS
+      messagesApp.addRecipient('a@b.c');
+      // Without waiting for the banner to disappear, return to thread list
+      messagesApp.performHeaderAction();
+      messagesApp.selectAppMenuOption('Discard');
+      // Wait a little bit, much less than the time needed for the banner to
+      // expire in order to give the render loop time to clear the banners.
+      client.helper.wait(300);
+      // Create another new message
+      threadList.navigateToComposer();
+
+      // The banner should not be displayed
+      assertIsNotDisplayed(messagesApp.Composer.conversionBanner);
+    });
+
+    test('MMS to email conversion and reminders', function() {
+      var composer = messagesApp.Composer;
+      var threadList = messagesApp.ThreadList;
+
+      // Case #1: When we open composer initially we should see not banner while
+      // we are in SMS mode.
+      threadList.navigateToComposer();
+
+      assertIsNotDisplayed(composer.mmsLabel);
+      assertIsNotDisplayed(composer.conversionBanner);
+
+      // Case #2: Add an email recipient and the message becomes MMS and the
+      // banner should appear.
+      messagesApp.addRecipient('a@b.c');
+      assertIsDisplayed(composer.mmsLabel);
+      assertIsDisplayed(composer.conversionBanner);
+
+      // Case #3: Wait for the banner to disappear, then send a message to
+      // be redirected to the thread view. We are still in MMS mode but no
+      // banner should be displayed.
+      waitForBannerToDisappear();
+      composer.messageInput.tap();
+      composer.messageInput.sendKeys('test');
+      messagesApp.send();
+
+      client.helper.waitForElement(composer.mmsLabel);
```

Comment:
Nop. It's not for checking we moved into thread list. It's simply to see if the MMS mode is kept enabled but as it is related with the MMS label and not the banner, maybe I can simply remove this line, What do you think?